src/ConversationList/ConversationListItem.vala

```diff
@@ -114,4 +114,45 @@ public class Mail.ConversationListItem : Gtk.ListBoxRow {
 
         return i;
     }
+
+    private string get_relative_datetime (DateTime date_time) {
+        var now = new DateTime.now_local ();
+        var diff = now.difference (date_time);
+
+        if (is_same_day (date_time, now)) {
+            if (diff < TimeSpan.MINUTE) {
+                return _("Now");
+            }
+            if (diff < TimeSpan.HOUR) {
+                return ngettext ("%dm ago", "%dm ago", (ulong) (diff / TimeSpan.MINUTE)).printf ((int) (diff / TimeSpan.MINUTE));
+            }
+            if (diff < 12 * TimeSpan.HOUR) {
+                int rounded = (int) Math.round ((double) diff / TimeSpan.HOUR);
```

Comment:
@tintou but `Math.round()` does not simply truncate. It rounds to the nearest integer, so 1.9 hours will be rounded to 2.